### PR TITLE
chore: clean up FIXME and TODO comments in our tests

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Clean up the `FIXME` and `TODO` comments in our tests

This enables a few tests and establishes a better pattern for platform-dependently skipping tests.

Notes: no-notes

### CODE REVIEW

File: spec/chromium-spec.js [hunk 1]
```diff
@@ -831,18 +831,9 @@ describe('chromium feature', () => {
...
-    })
-
     it('can be get as context in canvas', () => {
-      if (process.platform === 'linux') {
-        // FIXME(alexeykuzmin): Skip the test.
-        // this.skip()
-        return
-      }
+      // WebGL doesn't work on ia32 linux in CI
+      if (process.platform === 'linux' && process.arch === 'ia32' && isCI) return;
```
Comment (alexeykuzmin) [inline spec/chromium-spec.js]: The test should be marked as skipped, not as passed:
```js
if (...) {
  this.skip()
  return
}
```

File: spec/api-app-spec.js [hunk 2]
```diff
@@ -511,13 +510,7 @@ describe('app module', () => {
...
     })
 
-    it('allows you to pass a custom executable and arguments', function () {
-      if (process.platform !== 'win32') {
-        // FIXME(alexeykuzmin): Skip the test.
-        // this.skip()
-        return
-      }
-
+    platformIt('allows you to pass a custom executable and arguments', ['win32'], function () {
```
Comment (alexeykuzmin) [inline spec/api-app-spec.js]: Looks useful.
We also need `xplatformIt` and `xplatformDescribe` to match `xit` and `xdescribe` to temporary disable tests.

Comment (alexeykuzmin) [inline spec/api-app-spec.js]: Is `this` preserved inside a test function? To use `this.timeout()` for example.

File: spec/api-app-spec.js [hunk 3]
```diff
@@ -749,15 +742,6 @@ describe('app module', () => {
       large: process.platform === 'win32' ? 32 : 48
     }
 
-    // (alexeykuzmin): `.skip()` called in `before`
-    // doesn't affect nested `describe`s.
-    beforeEach(function () {
```
Comment (alexeykuzmin) [inline spec/api-app-spec.js]: Doesn't seem to be replaced by anything.
Is it no longer needed?

File: spec/api-app-spec.js (hunk 2 above)
Comment (MarshallOfSound) [reply spec/api-app-spec.js]: @alexeykuzmin Yes, mocha calls `fn.apply(mochaBIndings)` to make `this` work

**Comment by codebytere** [comment]
I started working on this a bit but a few concerns: this prevents us from running singular platform tests with `.only`. I'd prefer to simplify the platform-dependent tests we have in our codebase, but i think i'd prefer to do it in a way that doesn't obfuscate mocha for newcomers or make our specs more complicated without significant benefit.

**Comment by MarshallOfSound** [comment]
@codebytere 

> I started working on this a bit but have a concern: this prevents us from running singular platform tests with .only

This is not too hard to support, I've dumped a commit on the end here that should support `.only` and `.skip` --> https://github.com/electron/electron/pull/15782/commits/778c0eba03a3a0fc958e428025c91b2e7793bd71

**Comment by MarshallOfSound** [comment]
Although the spirit of this PR was good, it got left by the wayside and therefore is so conflicted it's not usable any more.

**Pull request closed without merging**